pkg/proxy/userspace/proxysocket.go

```diff
@@ -188,19 +188,19 @@ func (udp *udpProxySocket) Addr() net.Addr {
 }
 
 // Holds all the known UDP clients that have not timed out.
-type clientCache struct {
-	mu      sync.Mutex
-	clients map[string]net.Conn // addr string -> connection
+type ClientCache struct {
+	sync.Mutex
+	Clients map[string]net.Conn // addr string -> connection
```

Comment:
eh, there are situations where "lock, get, set if missing" is a bit nicer with the ability to directly lock.